CONTRIBUTING.md

```diff
@@ -1,19 +1,138 @@
-## Welcome!
-
-We're so glad you're thinking about contributing to an HHS IDEA Lab open source project! If you're unsure about anything, just ask -- or submit the issue or pull request anyway. The worst that can happen is you'll be politely asked to change something. We love all friendly contributions.
+# Welcome!
 
 We want to ensure a welcoming environment for all of our projects. Our staff follow the [18F Code of Conduct](https://github.com/18F/code-of-conduct/blob/master/code-of-conduct.md) and all contributors should do the same.
 
 We encourage you to read this project's CONTRIBUTING policy (you are here), its [LICENSE](LICENSE.md), and its [README](README.md).
 
-If you have any questions or want to read more, check out the [18F Open Source Policy GitHub repository]( https://github.com/18f/open-source-policy), which inspired this policy.
+Setting up the FHIR server locally
+---
+Clone the repository onto you're computer.
+```
+git clone https://github.com/CMSgov/beneficiary-fhir-data
+```
+
+The instructions from here on should be run from the `contributing` directory in repository.
+
+To simply run tests or execute other tasks in the BFD bring up the docker containers:
+```
+make up
+```
+
+This brings services up in the background and displays the logs from the `bfd` container. Once the logs show that that the system is started (this can take a minute or so depending on your machine) the logs can be exited with Ctrl+C (Cmd+C on a mac).
+
+Now the system can be interacted with. Here's an example of running tests for the `bfd-server` module.
+```
+docker-compose exec bfd bash
+cd /app/apps/bfd-server
+mvn verify
+```
+
+### Serving the BFD
+
+To run the BFD locally in a way that will allow you and other systems to interact with it some modifications need to be made so that it serves on a consistent port. These changes are contained in the file `contributing/patches/allow_local_port_config.patch` and can be applied with 
+
+```
+make servable
+```
+To undo the changes run `make unservable`.
+
+Since this changes the code in the repository please keep in mind not to commit these changes and to be aware of them while making your own changes.
```

Comment:
I suppose we wouldn't want to add these files to a `.gitignore`, in the event that someone actually needs to edit them.  Is there any other way we can protect the accidental commit of patched files?

Reply:
There is now work ticketed to make these patches unnecessary